types/applications/formDataExtended.d.ts

```diff
@@ -1,7 +1,50 @@
-declare class FormDataExtended extends FormData {}
-
 /**
- * @deprecated since 0.7.0
- * @see {@link FormApplication.processForm}
+ * An extension of the native FormData implementation.
+ *
+ * This class functions the same way that the default FormData does, but it is more opinionated about how
+ * input fields of certain types should be evaluated and handled.
+ *
+ * It also adds support for certain Foundry VTT specific concepts including:
+ *  Support for defined data types and type conversion
+ *  Support for TinyMCE editors
+ *  Support for editable HTML elements
+ *
  */
-declare function validateForm(formElement: HTMLFormElement): FormDataExtended;
+declare class FormDataExtended extends FormData {
+  /**
+   * @param form    - The form being processed
+   * @param editors - An array of TinyMCE editor instances which are present in this form
+   * @param dtypes  - A mapping of data types for form fields
+   */
+  constructor(
+    form: HTMLFormElement,
+    { editors, dtypes }: { editors?: FormDataExtended['editors']; dtypes?: FormDataExtended['dtypes'] }
+  );
+
+  /**
+   * A mapping of data types requested for each form field
+   * @defaultValue `{}`
+   */
+  dtypes: Record<string, string>;
+
+  /**
+   * A record of TinyMCE editors which are linked to this form
+   * @defaultValue `[]`
+   */
+  editors: FormDataExtended.EditorTuple[];
+
+  /**
+   * Process the HTML form element to populate the FormData instance.
+   * @param form - The HTML form
+   */
+  process(form: HTMLFormElement): void;
+
+  /**
+   * Export the FormData as an object
+   */
+  toObject(): Record<string, any>;
```

Comment:
What do you think about making this return `Record<string, unknown` instead? I don't think there is a ways to provide more concrete types than `any` / `unknown` and in that case, personally, I think `unknown` is the better choice. But then, it's the same old debate again. It's just that `any` suggests to me that it potentially could be improved, which is not the case here.